fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java

```diff
@@ -745,8 +747,32 @@ private ColumnStatistic getColumnStatistic(TableIf table, String colName, long i
                 return ColumnStatistic.UNKNOWN;
             }
         } else {
+            if (!partitionNames.isEmpty()) {
+                PartitionColumnStatisticBuilder builder = new PartitionColumnStatisticBuilder();
+                boolean isFirst = true;
+                boolean hasUnknown = false;
+                for (String partitionName : partitionNames) {
+                    PartitionColumnStatistic pcolStats = Env.getCurrentEnv().getStatisticsCache()
+                            .getPartitionColumnStatistics(
+                                    catalogId, dbId, table.getId(), idxId, partitionName, colName);
+                    if (pcolStats.isUnKnown) {
+                        hasUnknown = true;
+                        break;
+                    }
+                    if (isFirst) {
+                        builder = new PartitionColumnStatisticBuilder(pcolStats);
+                        isFirst = false;
+                    } else {
+                        builder.merge(pcolStats);
```

Comment:
do merge after check all pcol stats is not unknown to avoid useless merge?

Reply:
this check is done in line758

Reply:
delay merge after check all partition could get beter perf

Reply:
done